### PR TITLE
Fix XSS vulnerability

### PR DESCRIPTION
**Pull request by GDLMadushanka**
Add required encodings

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes/no
 - Ran FindSecurityBugs plugin and verified report? yes/no
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes/no

### CODE REVIEW

File: components/data-services/org.wso2.carbon.dataservices.ui/src/main/resources/web/ds/addOperation.jsp [hunk 1]
```diff
@@ -206,7 +206,7 @@
                 <th colspan="2"><fmt:message key="service.operations"/></th>
             </tr>
             </thead>
-            <form method="post" action="operationProcessor.jsp?action=<%=Encode.forHtmlAttribute(action)%>"
+            <form method="post" action="operationProcessor.jsp?action=<%=Encode.forHtml(action)%>"
```
Comment (thariyarox) [inline components/data-services/org.wso2.carbon.dataservices.ui/src/main/resources/web/ds/addOperation.jsp]: forUriComponent method should be used
https://owasp.github.io/owasp-java-encoder/encoder/apidocs/org/owasp/encoder/Encode.html#forUriComponent-java.lang.String-

File: components/data-services/org.wso2.carbon.dataservices.ui/src/main/resources/web/ds/addOperation.jsp [hunk 2]
```diff
@@ -343,7 +344,7 @@
                                     <td colspan="3">
                                         <a class="icon-link"
                                            style="background-image:url(../admin/images/add.gif);"
-                                           href='addOperationParameter.jsp?operationName=<%=Encode.forHtmlAttribute(operationName)%>&action=<%=action%>&queryId=<%=Encode.forHtmlAttribute(selectedQueryId)%>' ><fmt:message
+                                           href='addOperationParameter.jsp?operationName=<%=Encode.forHtmlAttribute(operationName)%>&action=<%=Encode.forHtml(action)%>&queryId=<%=Encode.forHtmlAttribute(selectedQueryId)%>' ><fmt:message
```
Comment (thariyarox) [inline components/data-services/org.wso2.carbon.dataservices.ui/src/main/resources/web/ds/addOperation.jsp]: forUriComponent should be used
https://owasp.github.io/owasp-java-encoder/encoder/apidocs/org/owasp/encoder/Encode.html#forUriComponent-java.lang.String-

File: components/data-services/org.wso2.carbon.dataservices.ui/src/main/resources/web/ds/addOperation.jsp [hunk 3]
```diff
@@ -334,7 +335,7 @@
                                     <td colspan="3">
                                         <a class="icon-link"
                                            style="background-image:url(../admin/images/add.gif);"
-                                           href='addOperation.jsp?param=qparam&operationName=<%=Encode.forHtmlAttribute(operationName)%>&operationDesc=<%=Encode.forHtmlAttribute(operationDesc) %>&action=<%=action%>&selectedQueryId=<%=Encode.forHtmlAttribute(selectedQueryId)%>' >Add Query Params as Operation Params</a>
+                                           href='addOperation.jsp?param=qparam&operationName=<%=Encode.forHtmlAttribute(operationName)%>&operationDesc=<%=Encode.forHtmlAttribute(operationDesc) %>&action=<%=Encode.forHtml(action)%>&selectedQueryId=<%=Encode.forHtmlAttribute(selectedQueryId)%>' >Add Query Params as Operation Params</a>
```
Comment (thariyarox) [inline components/data-services/org.wso2.carbon.dataservices.ui/src/main/resources/web/ds/addOperation.jsp]: forUriComponent should be used
https://owasp.github.io/owasp-java-encoder/encoder/apidocs/org/owasp/encoder/Encode.html#forUriComponent-java.lang.String-

File: components/data-services/org.wso2.carbon.dataservices.ui/src/main/resources/web/ds/addOperation.jsp [hunk 4]
```diff
@@ -239,7 +239,8 @@
...
                                     <td>
                                         <select name="queryId" id="queryId"
-                                                onchange="javascript:location.href = 'addOperation.jsp?selectedQueryId='+this.options[this.selectedIndex].value+'&operationName='+document.getElementById('operationName').value+'&operationDesc='+document.getElementById('operationDesc').value+'&enableStreaming='+document.getElementById('enableStreaming').value+'&returnRequestStatus='+document.getElementById('returnRequestStatus').value+'&flag=true&action=<%=action%>';">
+                                                onchange="javascript:location.href =
+                                                        'addOperation.jsp?selectedQueryId='+this.options[this.selectedIndex].value+'&operationName='+document.getElementById('operationName').value+'&operationDesc='+document.getElementById('operationDesc').value+'&enableStreaming='+document.getElementById('enableStreaming').value+'&returnRequestStatus='+document.getElementById('returnRequestStatus').value+'&flag=true&action=<%=Encode.forHtml(action)%>';">
```
Comment (NShani) [inline components/data-services/org.wso2.carbon.dataservices.ui/src/main/resources/web/ds/addOperation.jsp]: This should change to forUriComponent(action) method.
https://owasp.github.io/owasp-java-encoder/encoder/apidocs/org/owasp/encoder/Encode.html#forUriComponent-java.lang.String

File: components/data-services/org.wso2.carbon.dataservices.ui/src/main/resources/web/ds/addOperation.jsp [hunk 5]
```diff
@@ -309,7 +310,7 @@
                                                     <% if (!action.equals("")) { %>
                                                     <a class="icon-link"
                                                        style="background-image:url(../admin/images/edit.gif);"
-                                                       href='addOperationParameter.jsp?editparam=editparam&operationName=<%=Encode.forHtmlAttribute(operationName)%>&action=<%=action%>&queryId=<%=selectedQueryId%>&paramNameId=<%=aWithParamsList.getName()%>&operationParamId=<%=(aWithParamsList.getParamValue())%>'>
+                                                       href='addOperationParameter.jsp?editparam=editparam&operationName=<%=Encode.forHtmlAttribute(operationName)%>&action=<%=Encode.forHtml(action)%>&queryId=<%=selectedQueryId%>&paramNameId=<%=aWithParamsList.getName()%>&operationParamId=<%=(aWithParamsList.getParamValue())%>'>
```
Comment (thariyarox) [inline components/data-services/org.wso2.carbon.dataservices.ui/src/main/resources/web/ds/addOperation.jsp]: forUriComponent should be used
https://owasp.github.io/owasp-java-encoder/encoder/apidocs/org/owasp/encoder/Encode.html#forUriComponent-java.lang.String-

**Pull request merged**